File: code/bb247.cpp
#include <iostream>
using namespace std;
int main()
{
    int i,n, a[1000];
    cin>>n;
    for (i=1; i<=n; i++) cin>>a[i];
    int dem=0;
    for (i=1; i<=n; i++) {
        int s=0;
        for (int j=i; j<=n; j++) {
            s+=a[j];
            if (s==0) dem++;
        }
    }
    cout<<dem;
    return 0;
}

File: code/bb247.pas
uses math;
var i, n, dem, s, j: integer;
    a : array[1..1000] of integer;
BEGIN
    readln(n);
    for i:= 1 to n do read(a[i]);
    dem:=0;
    for i:= 1 to n do begin
        s:=0;
        for j:= i to n do begin
            s+=a[j];
            if s=0 then inc(dem);
        end;
    end;
    write(dem);
END.